### DISCUSSION TITLE
About parallel execution

### DISCUSSION THREAD

**Discussion by lnfel** [quirrel-dev/quirrel#1139] [Q&A]
I noticed the `exclusive` property at https://docs.quirrel.dev/api/queue#exclusive. Assuming `exclusive` is set to false and jobs are running in parallel (is it?), how do I know how many jobs are running at any given moment. For example I have 10 jobs in queue and all of them has no set `delay` (jobs will fire immediately), how do I see how many jobs are running out of 10 jobs I have?

**Comment by Skn0tt** [comment]
Hi @lnfel, excuse the late reply!

> how do I know how many jobs are running at any given moment

There's no way of finding that out right now, except for looking at the Quirrel UI and looking at the execution logs, i'm afraid. Could you elaborate on what your usecase for this would be?

**Reply by lnfel** [reply]
Good day @Skn0tt, really appreciate the reply. I've been working on [Aerial](https://github.com/lnfel/aerial) (an sveltekit app that extracts dominant colors on images and documents). It heavily relies on binary executable tools like [kmeans_colors](https://github.com/okaneco/kmeans-colors) (for extracting colors from images) and libreoffice (to convert document pages to images). I'm running it on a cheap AWS EC2 instance and running OOM is a very nasty experience. It would be nice to get info on how many instances of parallel jobs are being handled and setting allowed minimum number of parallel jobs would be key factor in ensuring the server won't crash. In any case, I could just set exclusive on a job so there won't be multiple instances of libreoffice.

Also there is quite an issue trying to make Quirrel work on production at #1154 .
